lib/crack.rb

```diff
@@ -1,18 +1,41 @@
 require_relative './enigma'
+require_relative './valid_input'
 
 ARGV == ['ecnrypted.txt', 'cracked.txt', 'date']
 ARGV[0] == 'encrypted.txt'
 ARGV[1] == 'cracked.txt'
 ARGV[2] == 'date'
 
-encrypted = File.open(ARGV[0], 'r')
-message = encrypted.read
-
+valid_input = ValidInput.new(ARGV[0], ARGV[1], ARGV[2])
 enigma = Enigma.new
-output = enigma.crack(message, ARGV[2])
 
-cracked = File.open(ARGV[1], 'w')
+encrypted_message = valid_input.check_valid_file
+
+if encrypted_message == 0
+  puts "\nThe file you are trying to load is not in the current directory \n \nDid you mean? \n"
+  Dir["./*.txt"].each { |file| puts '  -' + file[2..-1] }
+  exit
+end
+
+
+output_file = valid_input.check_valid_output_file_name
+
+if output_file == 0
+  puts "The file name you entered is not valid. Please enter a file name ending with '.txt'"
```

Comment:
prints a message to the user and exits the program if the text file name for output is not a valid text file name.